src/Engine/ProtoCore/Core.cs

```diff
@@ -2116,10 +1865,20 @@ public void GenerateExprExeInstructions(int blockScope)
             }
         }
 
+        private RuntimeData GenerateRuntimeData()
+        {
+            Validity.Assert(RuntimeData != null);
+            RuntimeData.FunctionTable = FunctionTable;
```

Comment:
This method mixes functional (returning the value) and state change (RuntimeData.FunctionTable = xxx) patterns.

This means that it's really surprising that it has both side effects and a return value. Lets pick one patten and stick to it.